sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -438,6 +438,22 @@ object DataSourceStrategy {
     }
   }
 
+  /**
+   * The attribute name may differ from the one in the schema if the query analyzer
+   * is case insensitive. We should change attribute names to match the ones in the schema,
+   * so we do not need to worry about case sensitivity anymore.
+   */
+  protected[sql] def normalizeExprs(
```

Comment:
I copied the body from `DataSourceStrategy$normalizeFilters` to minimize the changes. Obviously, it is a temp solution until we agree on a better way in the community. For example, I can submit a separate PR to rename `normalizeFilters`.

Reply:
+1 for the decision. Let's focus the new feature first.

Reply:
That's what I would propose as well. Here is [the PR](https://github.com/apache/spark/pull/26830).